src/ResourceManager/Kusto/Commands.Kusto/Commands/NewAzureRmKustoDatabase.cs

```diff
@@ -0,0 +1,147 @@
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 4,
+            Mandatory = true,
+            HelpMessage = "the amount of days.")]
+        [LocationCompleter("Microsoft.Kusto/databases")]
+        public int SoftDeletePeriodInDays { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Position = 5,
+            Mandatory = true,
+            HelpMessage = "Azure region where the cluster exists.")]
+        [LocationCompleter("Microsoft.Kusto/clusters")]
+        public int HotCachePeriodInDays { get; set; }
+
+        [Parameter(
+            ParameterSetName = ResourceIdParameterSet,
+            Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Kusto cluster ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = true,
+            Position = 2,
+            ValueFromPipeline = true,
+            HelpMessage = "Kusto cluster object.")]
+        [ValidateNotNullOrEmpty]
+        public PSKustoCluster InputObject { get; set; }
+
+
+        public override void ExecuteCmdlet()
+        {
+            string resourceGroupName = ResourceGroupName;
+            string clusterName = ClusterName;
+            string databaseName = Name;
+            string location = null;
+            if (ShouldProcess(Name, Resources.CreateNewKustoCluster))
+            {
+                try
+                {
+                    if (!string.IsNullOrEmpty(ResourceId))
+                    {
+                        KustoUtils.GetResourceGroupNameAndClusterNameFromClusterId(ResourceId, out resourceGroupName, out clusterName);
+                    }
+
+                    if (InputObject != null)
+                    {
+                        KustoUtils.GetResourceGroupNameAndClusterNameFromClusterId(InputObject.Id, out resourceGroupName, out clusterName);
+                    }
+
+                    var cluser = KustoClient.GetCluster(resourceGroupName, clusterName);
+                    if (cluser == null)
+                    {
+                        throw new CloudException(string.Format(Resources.KustoClusterNotExist, clusterName));
+                    }
+
+                    location = cluser.Location;
+
+                    var database = KustoClient.GetDatabase(resourceGroupName, clusterName, databaseName);
+                    if (database != null)
+                    {
+                        throw new CloudException(string.Format(Resources.KustoClusterExists, Name));
+                    }
+                }
+                catch (CloudException ex)
+                {
+                    if (ex.Body != null && !string.IsNullOrEmpty(ex.Body.Code) && ex.Body.Code == "ResourceNotFound" ||
```

Comment:
Won't this also throw if the Cluster is not found? And don't you want it to fail in that case?

Reply:
I prefer to let the command fail.